feat: add preventExcessiveDragging option to limit boundary slide gestures

aovcina opened the pull request:
**Implements:**
- #307

ismail9k commented:
Thank you, @aovcina, for your PR! I've reviewed it and found some edge cases where unusual behavior occurs during aggressive dragging. Instead of abruptly cutting off the drag action, I suggest preventing the track sliding from exceeding its thresholds by dynamically adjusting the `trackTransform` computed variable.

aovcina commented:
@ismail9k, thank you for the feedback. I've reviewed and tested made changes and you're right. It's better to handle it within the `trackTransform` computed. 
Along the way, I've added prop validation so developers are informed that the newly added `preventExcessiveDragging` option cannot work with `wrapAround`. Also, I've added a badge in docs for the upcoming package version for the `preventExcessiveDragging` config. Please amend if needed.
